src/core/network-openssl.c

```diff
@@ -648,6 +648,10 @@ static void set_pubkey_info(TLS_REC *tls, X509 *cert, unsigned char *cert_finger
 	bio = BIO_new(BIO_s_mem());
 	ASN1_TIME_print(bio, X509_get_notBefore(cert));
 	length = BIO_read(bio, buffer, sizeof(buffer));
+	if (length < 0) {
+		g_free(buffer);
+		return FALSE;
```

Comment:
This leaks cert_fingerprint_hex, public_key_fingerprint_hex, bio, pubkey, doing "goto error" is encouraged here. Also buffer is stack-allocated, doesn't need to be freed.

Same thing for the return FALSE below.

Reply:
Good points, I've just addressed your comments. Let me know if you want me to manually squash these commits.

Thank you for the review!